webdriver/webapp_server.go

```diff
@@ -126,146 +104,19 @@ func NewWebserver() (s AppServer, err error) {
-	}
-	s.cmd = exec.Command(
-		"dev_appserver.py",
-		"-A", s.project,
-		fmt.Sprintf("--port=%d", s.port),
-		// Let dev_appserver find free ports itself. We don't use the
-		// admin or API port directly.
-		"--api_port=0",
-		"--admin_port=0",
-		"--automatic_restart=false",
-		"--skip_sdk_update_check=true",
-		"--support_datastore_emulator=true",
-		"--clear_datastore=true",
-		"--datastore_consistency_policy=consistent",
-		fmt.Sprintf("--datastore_emulator_port=%d", s.gcdPort),
-		"--clear_search_indexes=true",
-		absAppYAMLPath,
-	)
-
-	log.Println("Starting dev_appserver")
-	for _, arg := range s.cmd.Args {
-		log.Println("  " + arg)
-	}
-
-	// dev_appserver.py usually does not print to stdout.
-	s.cmd.Stdout = os.Stderr
-	s.stderr, err = s.cmd.StderrPipe()
-	return s, err
-}
-
-var hostRE = regexp.MustCompile(`Starting module "default" running at: (\S+)`)
-var adminURLRE = regexp.MustCompile(`Starting admin server at: (\S+)`)
-var readyRE = regexp.MustCompile(`GET /_ah/warmup`)
-
-func (i *devAppServerInstance) AwaitReady() error {
-	if err := i.cmd.Start(); err != nil {
-		return err
-	}
-
-	// Read stderr until server is warmed up.
-	errc := make(chan error)
-	ready := false
-	go func() {
-		s := bufio.NewScanner(i.stderr)
-		defer i.stderr.Close()
-		for s.Scan() {
-			str := s.Text()
-			log.Println(str)
-			if match := readyRE.FindStringSubmatch(str); match != nil {
-				ready = true
-				errc <- nil
-				return
-			}
-			if match := hostRE.FindStringSubmatch(str); match != nil {
-				u, err := url.Parse(match[1])
-				if err != nil {
-					errc <- fmt.Errorf("failed to parse URL %q: %v", match[1], err)
-					return
-				}
-				i.baseURL = u
-			}
-			if match := adminURLRE.FindStringSubmatch(str); match != nil {
-				u, err := url.Parse(match[1])
-				if err != nil {
-					errc <- fmt.Errorf("failed to parse URL %q: %v", match[1], err)
-					return
-				}
-				i.adminURL = u
-			}
-		}
-		errc <- s.Err()
-	}()
-
-	exited := make(chan error)
-	go func() {
-		exited <- i.cmd.Wait()
-	}()
-
-	select {
-	case <-time.After(i.startupTimeout):
-		if p := i.cmd.Process; p != nil {
-			p.Kill()
-		}
-		return errors.New("timeout starting dev_appserver.py")
-	case err := <-errc:
-		if err != nil {
-			if p := i.cmd.Process; p != nil {
-				p.Kill()
-			}
-			return fmt.Errorf("error waiting for dev_appserver.py: %v", err)
-		}
-	case err := <-exited:
-		if err != nil {
-			return err
-		}
-	}
-
-	if !ready {
-		if p := i.cmd.Process; p != nil {
-			p.Kill()
-		}
-		return errors.New("dev_appserver.py unable to warm up")
-	}
-	if i.baseURL == nil {
-		return errors.New("unable to find webserver URL")
-	}
-	if i.adminURL == nil {
-		return errors.New("unable to find admin URL")
-	}
-	return nil
-}
-
 func addStaticData(i *devAppServerInstance) (err error) {
 	cmd := exec.Command(
 		"go",
 		"run",
 		"../util/populate_dev_data.go",
-		fmt.Sprintf("--project=%s", i.project),
-		fmt.Sprintf("--datastore_host=127.0.0.1:%d", i.gcdPort),
```

Comment:
Don't we still need this?

Reply:
No, `sharedtest.NewAEInstance` sets the env vars which will be picked up by `datastore.NewClient`.

https://github.com/web-platform-tests/wpt.fyi/blob/4720ce1316df824a536e0cf933c7ddfc82f64f55/shared/sharedtest/util.go#L91-L92

Reply:
yea you are right